lib/spack/spack/extensions.py

```diff
@@ -132,10 +137,31 @@ def load_extension(name: str) -> str:
 def get_extension_paths():
     """Return the list of canonicalized extension paths from config:extensions."""
     extension_paths = spack.config.get("config:extensions") or []
+    extension_paths.extend(get_extension_paths_from_entry_points())
     paths = [spack.util.path.canonicalize_path(p) for p in extension_paths]
     return paths
 
 
+def get_extension_paths_from_entry_points():
+    """Return the list of canonicalized extension paths from a project's
+    [project.entry-points."spack.extensions"]
+
+    """
+    extension_paths = []
+    if importlib_metadata is None:
+        return extension_paths
+    try:
+        entry_points = importlib_metadata.entry_points(group="spack.extensions")
+    except TypeError:
+        entry_points = importlib_metadata.entry_points().get("spack.extensions")
```

Comment:
This code and the import above are duplicated. They can probably be factored in a sepearte function and used both here and in `spack.config`.